### ISSUE TITLE
[Bug]: no-public-api-sidestep rule fails to handle path aliases without trailing slashes

### ISSUE THREAD

**Issue by rweapon** [effozen/eslint-plugin-fsd-lint#10]
### Describe the Bug
The `fsd/no-public-api-sidestep` rule incorrectly flags valid imports when using aliases without trailing slashes (e.g., `@entities` vs `@/entities`). The rule extracts the layer name correctly during initial parsing but fails to properly compare it against the import path parts during validation.

### Steps to Reproduce
1. Set up a project with FSD structure and path aliases (both with and without slashes)
2. Create an import like: `import { something } from "@entities/splitting"`
3. The linter incorrectly reports:  
   `🚨 Direct import from '@entities/splitting' is not allowed. Use the public API (index file) instead`

### Expected Behavior
The rule should:
1. Consistently handle both alias formats (`@entities` and `@/entities`)
2. Not flag valid imports when the public API is being used
3. Apply the same path parsing logic during validation as it does during initial layer extraction

### Technical Details
The issue occurs because:
1. Line 102 properly extracts the layer using `extractLayerFromImportPath()`
2. But lines 114-118 perform a strict equality check (`===`) against raw path parts
3. This fails when comparing parsed layer names against aliased paths

### Suggested Solutions
1. **Preferred**: Apply the same path normalization used in `extractLayerFromImportPath()` to the comparison logic
2. **Alternative**: Change strict equality (`===`) to `.includes()` check
3. **Workaround**: Currently requires manual layer configuration in rule options:
   ```json
   "fsd/no-public-api-sidestep": [
     "error", 
     { 
       "layers": ["@app", "@pages", "@widgets", "@features", "@entities", "@shared"] 
     }
   ]
   ```

The fix would make the rule more robust while maintaining its current validation purpose.

**Comment by effozen** [comment]
Good day, rweapon.
Thank you as always for your helpful bug reports — they truly help us improve the plugin.

Despite our efforts to consider every possible case and avoid bugs, they still sometimes happen.
I really appreciate your kindness in reporting the issue, even though it may have caused inconvenience.

I address it promptly and make sure the problem is resolved.
Once it's fixed, we’ll follow up with an update.

Thank you again for your continued support and thoughtful reports.
